stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -589,13 +598,17 @@ public class MessageListViewModel(
         messages.forEachIndexed { index, message ->
             val user = message.user
             val previousMessage = messages.getOrNull(index - 1)
+            val nextMessage = messages.getOrNull(index + 1)
 
             val previousUser = previousMessage?.user
-            val nextUser = messages.getOrNull(index + 1)?.user
+            val nextUser = nextMessage?.user
+
+            val willSeparateNextMessage =
+                nextMessage?.let { shouldAddDateSeparator(message, it) } ?: false
```

Comment:
Not directly related to this PR but the function `shouldAddDateSeparator()` uses a hardcoded time unit and value.

``kotlin
return timeDifference > TimeUnit.HOURS.toMillis(4)
```

Since we're giving users the ability to show or not show the date separators, would it also be nice to extract this time difference in a way that lets the user have more fine grained control over the date separators?

Reply:
Good point - I'll add a way to customize this logic